tests/core/test_task.py

```diff
@@ -51,6 +51,43 @@ class SampleTask(TaskBase):
             pass
 
 
+def test_task_decorator_can_have_iterable_output():
```

Comment:
maybe two test cases because this looks like a "happy path" test but also tests with the type error?